### ISSUE TITLE
pbr::MultiBar::listen blocks rayon threads

### ISSUE THREAD

**Issue by spease** [rayon-rs/rayon#565]
https://a8m.github.io/pb/doc/pbr/struct.MultiBar.html#method.listen

No matter how I run this (thread::spawn, rayon::scope, rayon::join) it seems to prevent into_par_iter from firing up multiple threads, which kind of defeats the point of having a multibar in the first place (as well as degrading performance).

Is there a right way to do this? Thanks.

**Comment by cuviper** [comment]
It's hard to guess without more context of what you're doing. Maybe the rayon threads are getting bottlenecked on writes to the individual progress bar pipes?

**Comment by spease** [comment]
It turned out to be a misattribution.

You have to generate the progress bars before you call listen and then use them, so I was putting them in a Vec.into_iter and then wrapping them in a mutex so I could then do progress_bars.lock().unwrap().next().unwrap() to extract them. However, I was doing this directly on a method call in a builder pattern, and the last step of that pattern was kicking off the operation. So the mutex lock inadvertently remained in scope and locked while the operation was running, while the other threads waited for the lock to release (which didn't happen until the operation completed).

Basically:

```
.par_iter()
.map(|...| {
    process_object
        .option(...)
        .option(...)
        .progress_bar(progress_bars.lock().unwrap().next().unwrap())
        .run(...)
})
```
